File: das_backend/src/main/java/ch/sbb/backend/admin/infrastructure/repositories/SpringDataJpaServicePointRepository.java
package ch.sbb.backend.admin.infrastructure.repositories;

import ch.sbb.backend.admin.infrastructure.model.ServicePointEntity;
import java.util.Optional;
import org.springframework.data.repository.ListCrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface SpringDataJpaServicePointRepository extends ListCrudRepository<ServicePointEntity, Integer> {

    Optional<ServicePointEntity> findByUic(int uic);
}

File: backend/src/main/kotlin/ch/sbb/backend/servicepoints/infrastructure/repositories/SpringDataJpaServicePointRepository.kt
package ch.sbb.backend.servicepoints.infrastructure.repositories

import ch.sbb.backend.servicepoints.infrastructure.entities.ServicePointEntity
import org.springframework.data.repository.ListCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface SpringDataJpaServicePointRepository : ListCrudRepository<ServicePointEntity, Int> {
     fun findByUic(uic: Int): ServicePointEntity?
}
